### DISCUSSION TITLE
I am trying to design the editor UI :)

### DISCUSSION THREAD

**Discussion by aappaapp** [VoylinsGamedevJourney/gozen#15] [Ideas]
Howdy! This project looks very cool! I am trying to design the UI for the editor, and maybe make a module. Any feedbacks?

Current unfinished design:

![Current unfinished design](https://github.com/voylin/GoZen/assets/35649969/e43ca936-fb90-40fb-965c-7778bc9f58d8)

**Comment by aappaapp** [comment]
Updated design (added some placeholder):

![Editor Design](https://github.com/voylin/GoZen/assets/35649969/a3228b45-1ab7-4ba3-9006-a65100d00dd6)

**Comment by voylin** [comment]
Main feedback which I can give at this point is where will the effects panel be? To change the composition, scaling, and other clip effects. I like the minimalistic design. Where would the menu buttons be for settings, and file control?

**Comment by aappaapp** [comment]
Another updated version! ^^
This time I added an inspector and a menu bar with a command palette. I completely forgot to add an inspector last time.

![Updated editor image](https://github.com/voylin/GoZen/assets/35649969/2219bfda-cafc-4269-8c73-d6eb96c4a0fa)

**Reply by voylin** [reply]
The debug tab would actually be quite handy if I could manage to implement it correctly ^^

**Comment by Wilker-uwu** [comment]
when i saw the very first screenshot, i thought that the whole thing was the mockup, which made me think of the familiarity and consistency that can make it easier to learn the program, meaning that we don't have to reinvent the wheel for this kind of program.

that said, most UIs like Premiere or Vegas are pretty cool, but there's always something in one that fails to be consistent in another, which isn't really good to just copy 1:1 into something intended to be as minimal as possible.

what i would suggest is that
(A) i think the imported files/file explorers don't have to take so much space if each piece of external media only gets imported once before being used throughout the project and its compositions,
(B) editing properties should take as much space as the video viewport does, since that's where the most time will be spent on, specially when some works involves making up tons of text and shapes on the spot without really importing anything, and,
(C) this is just personal, but i just think the zig-zag grids are annoying to navigate in programs like KDEnLive and Blender when two or more panels that are directly related to eachother in functionality are in opposing corners of the screen. maybe an audio monitor panel colliding on the right of the timeline panel is fine, but video imports and composition browsing should probably get columns of their own. that's why i like the default layout of Godot Engine, even though it's not perfect, but it's still pretty nice compared to a lot of video editors out there. avoiding zigzags should also make it more consistent to perhaps implement a keyboard-only navigation in the future.

![Godot engine window](https://godotengine.org/assets/features/editor-frame.png "Godot engine window")
![Somewhat of a mockup, using the origina Godot Engine window as a basis.](https://github.com/voylin/GoZen/assets/10854855/a549dd9b-3101-4289-9c81-0c147fefb1f6 "Somewhat of a mockup, using the origina Godot Engine window as a basis.")

###### some items noted in the mockup are but mentions of things that could appear in the future (i got carried away when making this), and nothing close to be ready to be implemented. in other words, beware of feature creep 😅

the picture below is an example of something similar in the software Enve Animation. not to scale.

![Example demonstration in Enve Animation](https://github.com/voylin/GoZen/assets/10854855/658e4e15-22a9-4deb-8fcb-1c7297c1ae7e "Example demonstration in Enve Animation")
